MAINTAINERS.md

```diff
@@ -22,3 +22,73 @@ We are grateful to our former maintainers for their contributions to the Jaeger
 * [@objectiser](https://github.com/objectiser)
 * [@tiffon](https://github.com/tiffon)
 * [@vprithvi](https://github.com/vprithvi)
+
+### Criteria for Maintainership
```

Comment:
are you repeating these from the Governance doc?

Reply:
i.e. we already have the process pretty well documented, the only part that was missing was the onboarding